sandboxify.c

```diff
@@ -39,75 +41,127 @@ int main(int argc, char **argv)
 	}
 
 	if (child) {
+		/* We created the child process */
 		int status = 0;
-		pid_t waitp = waitpid(child, &status, 0);
-
-		/* this should be executed after the first SIGSTOP from the child */
-		if (ptrace(PTRACE_SETOPTIONS, child, NULL, (TRACE_OPTS | PTRACE_O_SUSPEND_SECCOMP | PTRACE_O_EXITKILL))) {
-			perror("failed to suspend seccomp filters for the child");
-			/* PTRACE_O_EXITKILL was not yet set, so kill the child manually */
-			if (kill(child, SIGKILL)) {
-				perror("failed to kill the child process");
-			}
+		pid_t waitp;
+
+		/* Allow the child to use PTRACE on us */
+		if (prctl(PR_SET_PTRACER, child, 0, 0, 0) < 0) {
```

Comment:
We should probably ignore `EINVAL` here as it may just indicate that the current kernel does not have `CONFIG_SECURITY_YAMA` enabled

Reply:
Have you confirmed the return error?
I do not have a kernel with YAMA compiled out at hand, however simply disabling it with `echo 0 >/proc/sys/kernel/yama/ptrace_scope` does not make `PR_SET_PTRACER` return `EINVAL`.

Reply:
Means you do have YAMA compiled, because if YAMA is not compiled, the file `/proc/sys/kernel/yama/ptrace_scope` is absent and ptrace returns `EINVAL`

Reply:
got it, thanks for confirming. Added the check